File: TaxiMoUI/taximo_mobile/lib/user/models/review_dto.dart
class ReviewDto {
  final int reviewId;
  final int rideId;
  final int riderId;
  final int driverId;
  final double rating;
  final String? comment;
  final DateTime createdAt;

  ReviewDto({
    required this.reviewId,
    required this.rideId,
    required this.riderId,
    required this.driverId,
    required this.rating,
    this.comment,
    required this.createdAt,
  });

  factory ReviewDto.fromJson(Map<String, dynamic> json) {
    return ReviewDto(
      reviewId: json['reviewId'] as int,
      rideId: json['rideId'] as int,
      riderId: json['riderId'] as int,
      driverId: json['driverId'] as int,
      rating: (json['rating'] as num).toDouble(),
      comment: json['comment'] as String?,
      createdAt: DateTime.parse(json['createdAt'] as String),
    );
  }
}

class ReviewCreateDto {
  final int rideId;
  final int riderId;
  final int driverId;
  final double rating;
  final String? comment;

  ReviewCreateDto({
    required this.rideId,
    required this.riderId,
    required this.driverId,
    required this.rating,
    this.comment,
  });

  Map<String, dynamic> toJson() {
    return {
      'rideId': rideId,
      'riderId': riderId,
      'driverId': driverId,
      'rating': rating,
      if (comment != null && comment!.isNotEmpty) 'comment': comment,
    };
  }
}

File: TaxiMo/TaxiMo.Services/DTOs/ReviewDto.cs
namespace TaxiMo.Services.DTOs
{
    public class ReviewDto
    {
        public int ReviewId { get; set; }
        public int RideId { get; set; }
        public int RiderId { get; set; }
        public int DriverId { get; set; }
        public decimal Rating { get; set; }
        public string? Comment { get; set; }
        public DateTime CreatedAt { get; set; }
        
        // Optional fields for driver and rider info (for UI display)
        public string? UserName { get; set; }
        public string? UserPhotoUrl { get; set; }
        public string? UserFirstName { get; set; }
        public string? DriverName { get; set; }
        public string? DriverPhotoUrl { get; set; }
        public string? DriverFirstName { get; set; }
    }
}

